ibis/expr/api.py

```diff
@@ -1869,8 +1867,30 @@ def _timestamp_time(arg):
 # ---------------------------------------------------------------------
 # Time API
 
+def between_time(arg, lower, upper, timezone=None):
+    """
+    Check if the input expr falls between the lower/upper bounds
+    passed. Bounds are inclusive. All arguments must be comparable.
+
+    Parameters
+    ----------
+    lower : str, datetime.time
+    upper : str, datetime.time
+    timezone : str, timezone, default None
+
+    Returns
+    -------
+    is_between : BooleanValue
+    """
+
+    arg = arg.op().args[0]
```

Comment:
We can't do this for all allowed input types here. This needs to check that the input is a `Time` extraction operation. I'll merge #1143 which will provide a failing test case for this PR.